### DISCUSSION TITLE
HelmRelease : invalid signature for Flink Operator

### DISCUSSION THREAD

**Discussion by KomanRudden** [fluxcd/flux2#2822] [Q&A] [answered]
Hello all,

I have added Flink Operator to FluxCD and this has been deployed successfully, however, in the last few days there have been errors logged from the HelmRelease relating to `x509: certificate signed by unknown authority`

Before I give the details can I ask / state 
1. If I manually delete a HelmRelease from a cluster managed by Flux will it be redeployed (as well as it's dependants) by Flux after a reconcile ?
2. I am not quite sure if this is a Flux issue below

The certificate the operator created is as follows (two images below)
![image](https://user-images.githubusercontent.com/1030623/172617656-6891c4e8-ffe4-41c0-8d0c-fad7ec1417f5.png)

![image](https://user-images.githubusercontent.com/1030623/172617592-af9418c4-df1f-428a-aacf-357f76118019.png)

I have tried `flux suspend hr..` and `flux resume hr...` without success. 

![image](https://user-images.githubusercontent.com/1030623/172615919-32863538-4268-455c-a6de-50186153f1bd.png)

As I asked above if I manually delete the problem HelmRelease will Flux reconcile and deploy it again? I have tried `flux reconcile kustomization infrastructure ...` which is where I have my Flink Operator configured.

The HelmRepository I am using is https://downloads.apache.org/flink/flink-kubernetes-operator-0.1.0/
![image](https://user-images.githubusercontent.com/1030623/172617011-70953290-9a2e-4b3f-a7a0-f82956630bc3.png)

and the HelmRelease is 
![image](https://user-images.githubusercontent.com/1030623/172617101-746a4d9f-1ae5-4e87-a26b-bc68cb350fdf.png)

Thanks for your time

**Comment by stefanprodan** [comment, marked as answer]
If you've fixed the cert, you can tell Flux to retry the HelmRepository fetch with: `flux reconcile source helm flink-operator`.

**Reply by KomanRudden** [reply]
Ok, so I noticed this when I tried the command. Even though the HelmRepository is listed, Flux doesn't seem to know about it. Am I understanding this correctly?
![image](https://user-images.githubusercontent.com/1030623/172621304-1abc934c-7d94-4c54-9e6e-25ede3a23146.png)

Thanks again

**Reply by stefanprodan** [reply]
You need to update both Flux the CLI and the controllers on your cluster. Right now you have a CLI that is way newer than the controllers.

**Reply by KomanRudden** [reply]
Just to be sure I'm understanding (and for others that may read this) you mean this? So the CLI is much newer than the controllers?
![image](https://user-images.githubusercontent.com/1030623/172624371-6b088db2-32b8-4980-8829-862d11c35ada.png)
and this as well (my k8s version is old)
![image](https://user-images.githubusercontent.com/1030623/172625103-8b98167c-367c-422c-b8c9-47812cd0c1d3.png)

I'm going to try `flux install` now

**Reply by stefanprodan** [reply]
See here the upgrade procedure https://fluxcd.io/docs/installation/#upgrade

**Reply by KomanRudden** [reply]
Can I ask quickly before I do the upgrade if there will be any issues with my cluster still being at v1.19? Thanks

**Reply by stefanprodan** [reply]
Kubernetes 1.19 has reached end-of-life some time ago, we don't run e2e conformance tests on EOL versions, so Flux may hit bugs in the Kubernetes API that were fixed in newer versions.

**Reply by KomanRudden** [reply]
I managed to get Flux to update the link-operator to v1.0.0 from v0.1.0. The flink CRD was not upgraded though. If I want to upgrade the [Flink CRD](https://nightlies.apache.org/flink/flink-kubernetes-operator-docs-release-1.0/docs/operations/upgrade/#1-upgrading-the-crd) I am not quite sure how Flux handles this as the documentation is a manual process and all I have defined in my Flux project is the flink-operator. There is no CRD. The operator will create the CRD I assume but it doesn't seem to be doing any updates in the cluster. Thank you

**Reply by stefanprodan** [reply]
Helm doesn't upgrade CRDs, but Flux does. See here how to enable it: https://fluxcd.io/docs/components/helm/helmreleases/#crds

**Reply by KomanRudden** [reply]
Thank you Stefan. I tried both CreateReplace and Create and I am seeing this in the flink-operator helmrelease.

![image](https://user-images.githubusercontent.com/1030623/173043078-1e36129c-23f9-46d6-85a1-c4311753d69e.png)

When I tested this on a local cluster I manually edited the CRD and added the missing version from the CRD definition and this seemed to work but it seems like I shouldn't be updating this manually? (line 16 & 18194 below)
![image](https://user-images.githubusercontent.com/1030623/173043633-d67b700b-f9f7-4d76-81ff-2b6ea147791d.png)

**Reply by stefanprodan** [reply]
Did you applied those manually on the cluster at some point?

**Reply by stefanprodan** [reply]
I think you need to take this up with the Flink team, if the CRDs are rejected by Kubernetes API there is nothing we can do in Flux about it.

**Reply by KomanRudden** [reply]
Yup, I think you're right. Flux is doing what it is supposed to. Thanks for the tip on the CreateReplace
